ydb/core/fq/libs/row_dispatcher/format_handler/filters/filters_set.cpp

```diff
@@ -8,236 +10,238 @@ namespace {
-        InFlightCompilations.erase(requestIt);
+        Counters_.InFlightCompileRequests->Inc();
+        return iter->second;
+    }
 
-        const auto filterIt = Filters.find(filterId);
-        if (filterIt == Filters.end()) {
-            LOG_ROW_DISPATCHER_DEBUG("Compile response ignored for id " << ev->Cookie << ", filter with id " << filterId << " not found");
+    void AbortCompileProgram(NActors::TActorId clientId) {
+        const auto iter = CompileHandlers_.find(clientId);
+        if (iter == CompileHandlers_.end()) {
             return;
         }
 
-        filterIt->second.OnCompileResponse(std::move(ev));
+        for (auto& [name, compileHandler] : iter->second) {
+            compileHandler->AbortCompilation();
+            const auto cookie = compileHandler->GetCookie();
+            InFlightCompilations_.erase(cookie);
+            CompileHandlers_.erase(iter);
+            Counters_.InFlightCompileRequests->Dec();
+        }
     }
 
-    TStatus AddFilter(IFilteredDataConsumer::TPtr filter) override {
-        LOG_ROW_DISPATCHER_TRACE("Create filter with id " << filter->GetFilterId());
+    TValueStatus<IProgramCompileHandler::TPtr> RemoveCompileProgram(ui64 cookie) {
+        const auto requestIter = InFlightCompilations_.find(cookie);
+        if (requestIter == InFlightCompilations_.end()) {
+            return TStatus::Fail(EStatusId::INTERNAL_ERROR, TStringBuilder() << "Compile response ignored for id " << cookie);
+        }
 
-        IPurecalcFilter::TPtr purecalcFilter;
-        if (const auto& predicate = filter->GetWhereFilter()) {
-            LOG_ROW_DISPATCHER_TRACE("Create purecalc filter for predicate '" << predicate << "' (filter id: " << filter->GetFilterId() << ")");
+        const auto [clientId, name] = requestIter->second;
 
-            auto filterStatus = CreatePurecalcFilter(filter);
-            if (filterStatus.IsFail()) {
-                return filterStatus;
-            }
-            purecalcFilter = filterStatus.DetachResult();
+        const auto handlersIter = CompileHandlers_.find(clientId);
+        if (handlersIter == CompileHandlers_.end()) {
+            return TStatus::Fail(EStatusId::INTERNAL_ERROR, TStringBuilder() << "Compile response ignored for id " << cookie << ", program with client id " << clientId << " not found");
         }
 
-        const auto [it, inserted] = Filters.insert({filter->GetFilterId(), TFilterHandler(*this, filter, std::move(purecalcFilter))});
-        if (!inserted) {
-            return TStatus::Fail(EStatusId::INTERNAL_ERROR, TStringBuilder() << "Failed to create new filter, filter with id " << filter->GetFilterId() << " already exists");
+        const auto iter = handlersIter->second.find(name);
+        if (iter == handlersIter->second.end()) {
+            return TStatus::Fail(EStatusId::INTERNAL_ERROR, TStringBuilder() << "Compile response ignored for id " << cookie << ", program with client id " << clientId << " not found");
```

Comment:
Тут тоже можно добавить name

```suggestion
            return TStatus::Fail(EStatusId::INTERNAL_ERROR, TStringBuilder() << "Compile response ignored for id " << cookie << ", program with client id " << clientId << " and name " << name << " not found");
```